src/query/bm25.rs

```diff
@@ -2,11 +2,53 @@ use serde::{Deserialize, Serialize};
 
 use crate::fieldnorm::FieldNormReader;
 use crate::query::Explanation;
+use crate::schema::Field;
 use crate::{Score, Searcher, Term};
 
 const K1: Score = 1.2;
 const B: Score = 0.75;
 
+/// An interface to compute the statistics needed in BM25 scoring.
+///
+/// The standard implementation is a [Searcher] but you can also
+/// create your own to adjust the statistics.
+pub trait Bm25StatisticsProvider {
+    /// The total number of tokens in a given field across all documents in
+    /// the index.
+    fn total_num_tokens(&self, field: Field) -> crate::Result<u64>;
```

Comment:
Doesn't putting `total_num_tokens` and `total_num_docs` into separate methods mean that we access at least the metadata of the inverted indexes for all segments twice now instead of once as without the trait? Does this not have a measurable effect on performance? Would it suffice to provide these values in a single method, e.g.

```rust
fn total_numbers(&self, field: Field) -> crate::Result<(u64, u64)>;
```

Reply:
@adamreichold We all like performance, but this has no impact.